src/runtime/action/index.ts

```diff
@@ -0,0 +1,8 @@
+export interface ActionReturn<Parameters = any> {
+	update?: (parameters: Parameters) => void;
+	destroy?: () => void;
+}
+
+export interface Action<Parameters = any, Element = HTMLElement> {
```

Comment:
After going back and forth on this, I think it's better to have `Element` as the first generic, to have the same order as the parameters. It just feels very unintuitive in my head the other way around.

Reply:
Tbh, I like the way it is with the option to narrow down the element as an optional. Imho this reflects the common usage-scenario of actions quite well. But it is a matter of taste and explicit vs implicit logic.